ONSdigital/eq-author-app
# EAR-1884 - use dates routing logic

### What is the context of this PR?

https://jira.ons.gov.uk/browse/EAR-1884

### How to review

- Create a new questionnaire, for the first question, add a date answer.
- Add one or more questions in the same folder as above.
- Create a duplicate of the first section.
- For the first date answer question, create a routing logic.
- In the routing logic, select the date answer.
- Choose a condition, 'Earlier than' / 'Later than', and enter a number value in the 'years' field.
- Then choose 'Before' or 'After' in the second dropdown.
- In the 'Then' field, select the destination as 'Go To end of current section'. Keep the 'Else' as 'Go To next page'.
- Run the survey, enter the date answer and check if the routing logic works correctly.
- For example, if the routing condition is 'Date answer is Earlier Than 5 years Before response date', then 'Go To end of current section'. If the Date answer is 05/05/2005, that is, earlier than 5 years before today, the survey must skip the questions after the date question and go to the end of the first section. If the Date answer is 10/10/2020, the survey must go to the next page.

### What to do after everything is green

1. - [ ] Bring this branch up-to-date with Origin/Master
2. - [ ] If there are a lot of commits or some are not helpful to read, squash them down
3. - [ ] Click the **Merge** button on this pull request
4. - [ ] Does this change mean the **Capability examples** questionnaire should be updated?
5. - [ ] Move the Jira ticket for this task into the next stage of the process
